odoo/tools/translate.py

```diff
@@ -443,8 +443,18 @@ def _get_lang(self, frame):
                     lang = env['res.users'].context_get()['lang']
         return lang
 
-    def __call__(self, source):
-        return self._get_translation(source)
+    def __call__(self, source, *args, **kwargs):
+        translation = self._get_translation(source)
+        assert not (args and kwargs)
+        if args or kwargs:
+            try:
+                return translation % (args or kwargs)
+            except (TypeError, ValueError, KeyError):
+                bad = translation
+                # fallback on source
+                translation = source % (args or kwargs)
```

Comment:
```suggestion
                # fallback: apply to source before logging exception (in case source fails)
                translation = source % (args or kwargs)
```